oak_client/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub mod proto {
+    #![allow(clippy::return_self_not_must_use)]
+    tonic::include_proto!("oak.session.noninteractive.v1");
+}
+
+pub mod transport;
+pub mod verifier;
+
+use crate::{
+    transport::AsyncTransport,
+    verifier::{EvidenceProvider, ReferenceValue, Verifier},
+};
+use anyhow::Context;
+use std::{vec, vec::Vec};
+
+pub struct OakClientBuilder {
+    transport: Box<dyn AsyncTransport>,
+    evidence_provider: Box<dyn EvidenceProvider>,
+    reference_value: ReferenceValue,
+    verifier: Box<dyn Verifier>,
+    crypto_provider: CryptoProvider,
+}
+
+impl OakClientBuilder {
```

Comment:
not sure I understand the point of the builder struct, if it's just created and then immediately used to build a cient. Isn't it simpler to just have a single function?

Reply:
Done
Moved it to `OakClient::create(...)`